src/clj_kondo/impl/utils.clj

```diff
@@ -142,7 +142,8 @@
   ([])
   ([a] a)
   ([a b]
-   (cond (and (map? a) (map? b))
+   (cond (:replace (meta b)) b
```

Comment:
the `:replace` lookup can be prevented when there is no meta

Reply:
Is `some->` a good option?

Reply:
This seems a tiny bit faster:

``` clojure
user=> (time (dotimes [i 100000000] (when-let [m (meta 1)] (:replace m))))
"Elapsed time: 1561.475506 msecs"
```